src/averageAges.js

```diff
@@ -55,7 +63,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const results = onlyWithSon
+    ? people.filter(data => data.sex === 'm'
+      && people.some(mother => mother.name === data.mother))
+    : people.filter(data =>
+      people.some(mother => mother.name === data.mother));
```

Comment:
In both cases you filter the people. You can just move all the check inside callback of one filter method